Source/Particles/Filter/FilterFunctors.H

```diff
@@ -100,9 +108,15 @@ struct ParserFilter
         amrex::Real const x  = p.pos(0);
         amrex::Real const y  = p.pos(1);
         amrex::Real const z  = p.pos(2);
-        amrex::Real const ux = p.rdata(PIdx::ux)/PhysConst::c;
-        amrex::Real const uy = p.rdata(PIdx::uy)/PhysConst::c;
-        amrex::Real const uz = p.rdata(PIdx::uz)/PhysConst::c;
+        amrex::Real ux = p.rdata(PIdx::ux)/PhysConst::c;
+        amrex::Real uy = p.rdata(PIdx::uy)/PhysConst::c;
+        amrex::Real uz = p.rdata(PIdx::uz)/PhysConst::c;
+        if (m_units == InputUnits::SI)
+        {
+            ux /= m_mass;
+            uy /= m_mass;
+            uz /= m_mass;
+        }
         if ( m_function_partparser(m_t,x,y,z,ux,uy,uz) > 0.5 ) return 1;
```

Comment:
Maybe useful to add:
```suggestion
        // ux, uy, uz are now in beta*gamma
        if ( m_function_partparser(m_t,x,y,z,ux,uy,uz) > 0.5 ) return 1;
```